report_worker/registry.py

```diff
@@ -0,0 +1,164 @@
+"""
+Report Registry
+
+This module provides a registry system for different report generators.
+Each report type has a unique key and associated generator function.
+"""
+
+import logging
+from typing import Dict, Callable, Any, Optional
+from abc import ABC, abstractmethod
+
+class ReportGeneratorBase(ABC):
+    """Base class for all report generators"""
+    
+    @abstractmethod
+    async def generate(self, job_id: str, organization_id: str, parameters: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Generate a report and return metadata about the generated artifact.
+        
+        Args:
+            job_id: Unique identifier for the report job
+            organization_id: Organization requesting the report
+            parameters: Report-specific parameters
+            
+        Returns:
+            Dict containing:
+            - blob_url: URL to the generated report artifact
+            - file_name: Name of the generated file
+            - file_size: Size of the file in bytes
+            - content_type: MIME type of the generated file
+            - metadata: Any additional metadata about the report
+        """
+        pass
+
+class SampleReportGenerator(ReportGeneratorBase):
+    """Sample report generator for demonstration purposes"""
+    
+    async def generate(self, job_id: str, organization_id: str, parameters: Dict[str, Any]) -> Dict[str, Any]:
+        """Generate a sample report"""
+        import json
+        from datetime import datetime
+        from azure.storage.blob import BlobServiceClient
```

Comment:
@Alexjes98 Microsoft’s guidance is to reuse client instances across invocations (don’t create them per request)

so Reuse clients across invocations + async SDK

Fix (best practice)

- Centralize a single async BlobServiceClient per Python worker in shared/blob_client_async.py.
- Use azure.storage.blob.aio in async code.